app/api/userdata/route.js

```diff
@@ -27,22 +27,52 @@ export async function GET(req) {
             const docRef = doc(db, collection, document);
             const docSnap = await getDoc(docRef);
             const userData = docSnap.data();
-            return NextResponse.json( userData )
+            return NextResponse.json(userData)
         }
-        
-    } catch(error) {
-        return NextResponse.json( error )
-    }       
+
+    } catch (error) {
+        return NextResponse.json(error)
+    }
 }
 
 export async function POST(req) {
     const userData = await req.json();
     try {
-        if (userData.action==="register") {
+        if (userData.action === "register") {
             await setDoc(doc(db, "userdata", userData.uid), userData.newUserData);
             return NextResponse.json()
         };
-    } catch(error) {
-        return NextResponse.json( error )
+    } catch (error) {
+        return NextResponse.json(error)
+    }
+}
+
+export async function PATCH(req) {
+    const newData = await req.json();
+    try {
+        if (newData.field === 'First name') {
+            const userRef = doc(db, "userdata", newData.uid);
+            await updateDoc(userRef, {
+                first: newData.newVal,
+            });
+        } else if (newData.field === 'Last name') {
+            const userRef = doc(db, "userdata", newData.uid);
+            await updateDoc(userRef, {
+                last: newData.newVal,
+            });
+        } else if (newData.field === 'Gender') {
+            const userRef = doc(db, "userdata", newData.uid);
+            await updateDoc(userRef, {
+                gender: newData.newVal,
+            });
+        } else if (newData.field === 'Date of birth') {
+            const userRef = doc(db, "userdata", newData.uid);
+            await updateDoc(userRef, {
+                date: newData.newVal,
+            });
+        }
+        return NextResponse.json('success')
+    } catch (error) {
+        return NextResponse.json(error)
```

Comment:
wait does this return the error as a 200 http status? if that's the case, we might not want to do that.

Reply:
No it returns 'success' if the code inside of the try block finishes without errors. If there is an error, it returns the error

Reply:
Yes i see that. But what is the http status code that is being returned in the error scenario?

Reply:
maybe idk how next likes to set up the api routes. but typically if you make an API call and it fails, you want to return the appropriate http status code. https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#client_error_responses
if you just return a 200 http status code with json in it, that means that the client is responsible for parsing the response and checking that the json says `success`.

Reply:
No, you are right, I see what you mean now, I am going to fix it in the next commit